diff --git a/core/Cargo.toml b/core/Cargo.toml
index d8f44e4..d68e106 100644
--- a/core/Cargo.toml
+++ b/core/Cargo.toml
@@ -38,7 +38,7 @@ void = "1"
 zeroize = "0.9"
 
 [target.'cfg(not(any(target_os = "emscripten", target_os = "unknown")))'.dependencies]
-ring = { version = "0.14", features = ["use_heap"], default-features = false }
+ring = { version = "^0.16", features = ["alloc"], default-features = false }
 untrusted = { version = "0.6" }
 
 [dev-dependencies]
diff --git a/protocols/noise/Cargo.toml b/protocols/noise/Cargo.toml
index 189c61b..803c418 100644
--- a/protocols/noise/Cargo.toml
+++ b/protocols/noise/Cargo.toml
@@ -16,8 +16,8 @@ libp2p-core = { version = "0.12.0", path = "../../core" }
 log = "0.4"
 protobuf = "2.3"
 rand = "0.6.5"
-ring = { version = "0.14", features = ["use_heap"], default-features = false }
-snow = { version = "0.5.2", features = ["ring-resolver"], default-features = false }
+ring = { version = "^0.16", features = ["alloc"], default-features = false }
+snow = { version = "0.6.1", features = ["ring-resolver"], default-features = false }
 tokio-io = "0.1"
 x25519-dalek = "0.5"
 zeroize = "0.9"
diff --git a/protocols/secio/Cargo.toml b/protocols/secio/Cargo.toml
index c65d13b..e7121e4 100644
--- a/protocols/secio/Cargo.toml
+++ b/protocols/secio/Cargo.toml
@@ -27,7 +27,7 @@ sha2 = "0.8.0"
 hmac = "0.7.0"
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
-ring = { version = "0.14", features = ["use_heap"], default-features = false }
+ring = { version = "^0.16", features = ["alloc"], default-features = false }
 untrusted = { version = "0.6" }
 
 [target.'cfg(target_arch = "wasm32")'.dependencies]
diff --git a/transports/websocket/Cargo.toml b/transports/websocket/Cargo.toml
index 026fbd5..35941c3 100644
--- a/transports/websocket/Cargo.toml
+++ b/transports/websocket/Cargo.toml
@@ -17,10 +17,10 @@ log = "0.4.1"
 rw-stream-sink = { version = "0.1.1", path = "../../misc/rw-stream-sink" }
 tokio-codec = "0.1.1"
 tokio-io = "0.1.12"
-tokio-rustls = "0.10.0-alpha.3"
+tokio-rustls = "0.10.1"
 soketto = { version = "0.2.3", features = ["deflate"] }
 url = "1.7.2"
-webpki-roots = "0.16.0"
+webpki-roots = "0.17.0"
 
 [dev-dependencies]
 libp2p-tcp = { version = "0.12.0", path = "../tcp" }
